docs/diagnostics.md

```diff
@@ -0,0 +1,44 @@
+# Diagnostics and Troubleshooting
+This document gives brief references to diagnostics and troubleshooting
+interfaces available in the library.
+
+A small collection of print routines is available which can be used to examine
+key data strucures, such as the SplinterDB trunk pages, BTree pages, space-allocation
+maps and so on.
+
+- [trunk_print_super_block](../src/trunk.c#:~:text=trunk\_print\_super\_block\(trunk\_handle)
+  Print contents of SplinterDB super block
+
+- [trunk_print](../src/trunk.c#:~:text=trunk\_print\_node\(trunk\_handle)
+  Print contents of a Trunk node (page)
+
+- [trunk_sprint_subtree](../src/trunk.c#:~:text=trunk\_sprint\_subtree\(platform\_stream\_handle)
+  Print contents of all sub-trees under a Trunk node (page)
+
+- [trunk_print_space_use](../src/trunk.c#:~:text=trunk\_print\_space\_use\(trunk\_handle)
+  Print space usage by level across all trunk nodes
+
+- [trunk_sprint_memtable](../src/trunk.c#:~:text=trunk\_sprint\_memtable\(platform\_stream\_handle)
+  Print contents of all Memtables in Splinter. Drills-down to printing BTrees under Memtables
+
+- [btree_print_node](../src/btree.c#:~:text=btree\_print\_node\(cache),
```

Comment:
Should probably also mention `btree_print_tree` and maybe `btree_print_lookup`.

Reply:
Done. Added.